api/src/main/java/org/openmrs/module/registrationapp/converter/RegistrationSummaryExtensionsGenerator.java

```diff
@@ -0,0 +1,116 @@
+	 * @param distribute flag to distribute summary widgets into columns or not
+	 * @return The list of registration summary widgets extensions.
+	 * 
+	 * @throws IllegalArgumentException
+	 * 
+	 * @should throw when the provided app is not a registration app
+	 */
+	public static List<Extension> generate(AppDescriptor app, boolean distribute) throws IllegalArgumentException {
+		
+		if (!isRegAppConfig(app)) {
+			throw new IllegalArgumentException("Not a Registration application configuration.");
+		}
+		
+		List<Extension> extensions = new ArrayList<Extension>();
+		JsonNode sections = app.getConfig().get("sections");
+		
+		if (sections != null && sections.size() > 0 ) {
+			boolean firstColumn = false;
+			for (JsonNode section : sections) {
+				String sectionId = section.get("id").getTextValue();
+				if ("contactInfo".equalsIgnoreCase(sectionId) || "demographics".equalsIgnoreCase(sectionId)){
+					// 'contact info' and 'demographics' are hardcoded in the reg. summary dashboard view
+					continue;
+				}
+				
+				String appId = app.getId();
+				String regSummaryId = generateRegSummaryId(appId, section.get("id").getTextValue());
+				String extensionPointId = (firstColumn = !firstColumn) || !distribute ? "registrationSummary.contentFragments" 
+				                                                                      : "registrationSummary.secondColumnContentFragments";
+				
+				Map<String, Object> extensionParams = new HashMap<String, Object>();
+				Map<String, String> fragmentConfig = new HashMap<String, String>();
+				fragmentConfig.put("sectionId", section.get("id").getTextValue());
+				
+				extensionParams.put("provider", "registrationapp");
+				extensionParams.put("fragment", "summary/section");
+				extensionParams.put("fragmentConfig", fragmentConfig);
+			
+				Extension extension = new Extension(regSummaryId
+													,appId
+													,extensionPointId
+													,null, null, null, 1, null
+													,extensionParams);
+				extensions.add(extension);	
+			}
+		}
+		return extensions;
+	}
+
+	/**
+	 * @return true if the app is a registration app, false otherwise
+	 */
+	public static boolean isRegAppConfig(AppDescriptor app) {
+		return (app.getInstanceOf() != null && "registrationapp.registerPatient".equals(app.getInstanceOf()));
+	}
+	
+	/**
+	 * Generates a comprehensive registration summary ID from an app ID and section ID.
+	 */
+	public static String generateRegSummaryId(String regAppId, String sectionId) {
+		String regSummaryId = regAppId.substring(0, regAppId.indexOf("registerPatient")) 
+								+ "summary." 
+								+ sectionId;
+		return regSummaryId;
+	}
+	
+	/**
+	 * Extracts extensions with the specified extensionPointId
+	 * 
+	 * @param extensions list summary extensions
+	 * @param extensionPointId
+	 * @return Extensions
+	 */
+	public static List<Extension> extractExtensions(List<Extension> extensions, String extensionPointId) {
```

Comment:
Since you can't use Lambda expressions (sigh), can you look at `CollectionUtils.filter(..)`?

You will most likely not need this method anymore then.

Reply:
Actually this module still depends on java 1.6 that doesn't support lambda expressions. Going ahead to `CollectionUtils.filter(..)`. Thanks @mks-d .